### ISSUE TITLE
Exception thrown when freeing message handle in SocketProxy

### ISSUE THREAD

**Issue by ghost** [zeromq/clrzmq#69]
Hi,
I'm using the latest clrzmq (3.0.0 alpha 1) with zmqlub 2.2.0 (on WinXP SP3 32-bit) for a prototype and cannot get a SUB socket to dispose properly. An exception is thrown in SocketProxy when trying to dispose the handle to the message buffer.  Exception is: The handle is invalid. (Exception from HRESULT: 0x80070006 (E_HANDLE)).

The code in question is:

```
 protected virtual void Dispose(bool disposing)
 {
     if (!_disposed)
     {
        if (_message != IntPtr.Zero)
        {
           Marshal.FreeHGlobal(_message);
           _message = IntPtr.Zero;
        }

        Close();
    }

   _disposed = true;
 }
```

Any ideas as to why this is happening? I don't think it would be safe to ignore the exception as it will lead to a memory leak.

Thanks,
V.

**Comment by jgoz** [comment]
A few questions for you:
1. Just to confirm, this exception gets thrown on the `Marshal.FreeHGlobal(_message)` line?
2. Does this only happen with SUB sockets?
3. Is the SUB socket being used in multiple threads?
4. Can you post some of your code that shows how the socket is being used & disposed?

Thanks. Those will help me find the problem.

**Comment by jgoz** [comment]
I think this is related to issue #77.

**Comment by jgoz** [comment]
I'm going to close this issue as it may have been fixed with issue #77. Please update your sources and try again.
